src/js/game/LevelMVC/LevelView.js

```diff
@@ -1159,6 +1159,62 @@ module.exports = class LevelView {
     }
   }
 
+  refreshActionPlane(levelData, indices) {
+    var sprite,
+      blockType;
+
+    for (var index in indices) {
+      if (indices[index] >= 0) {
+        let workingIndex = indices[index];
+
+        sprite = this.actionPlaneBlocks[workingIndex];
+        this.actionPlane.remove(sprite);
+
+        let y = Math.floor(workingIndex / this.controller.levelModel.planeHeight);
+        let x = workingIndex - (y * this.controller.levelModel.planeHeight);
+
+        let blockIndex = (this.yToIndex(y)) + x;
+
+        sprite = null;
+        if (!levelData.actionPlane[blockIndex].isEmpty) {
+          blockType = levelData.actionPlane[blockIndex].blockType;
+          sprite = this.createBlock(this.actionPlane, x, y, blockType, levelData);
+          if (sprite !== null) {
+            sprite.sortOrder = this.yToIndex(y);
+          }
+        }
+        this.actionPlaneBlocks[workingIndex] = sprite;
+      }
+    }
+
+    // Save all all non-block elements.
+    let entityHolder = [];
+    for (var aPIndex in this.actionPlane.children) {
+      let workingKey = this.actionPlane.children[aPIndex].key;
+      if (workingKey !== "blocks" && workingKey !== "torch" && workingKey !== "door"){
+        entityHolder.push(this.actionPlane.children[aPIndex]);
+      }
+    }
+
+    // Empty the actionPlane.
+    this.actionPlane.children = [];
```

Comment:
The problem I encountered with this is that actionPlane has elements numbered sequentially, so if you remove a sprite from the middle (say actionPlaneBlocks[index] corresponds to actionPlane[6] or something), then you'll get a gap in actionPlane.

example:
actionPlane = {0: <sprite>, 1: <sprite>, 3: <sprite>, ......};

what -that- does it kill phaser.js preUpdate/postUpdate/Update functions, because they are expecting linearly numbered indicies. It loops through sprites to draw with for (let i = 0; i < size; ++i). Thus when it hits '2' in the example above, it errors.

The dumping and repushing is so that everything in view.actionPlane remains numbered properly.